test/unit/colorspace_spec.js

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2017 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ColorSpace } from '../../src/core/colorspace';
+
+describe('colorspace', function () {
+  it('should be true if decode is not an array', function() {
+    expect(ColorSpace.isDefaultDecode('I am a string', 0)).toBe(true);
+    expect(ColorSpace.isDefaultDecode(undefined, 0)).toBe(true);
+    expect(ColorSpace.isDefaultDecode({}, 0)).toBe(true);
+  });
+  it('should be true if length of decode array is not correct', function() {
+    expect(ColorSpace.isDefaultDecode([0], 1)).toBe(true);
+    expect(ColorSpace.isDefaultDecode([0, 1, 0], 1)).toBe(true);
+  });
+  it('should be true if decode map matches the default decode map', function() {
+    expect(ColorSpace.isDefaultDecode([], 0)).toBe(true);
+    expect(ColorSpace.isDefaultDecode([0, 0], 1)).toBe(false);
+    expect(ColorSpace.isDefaultDecode([0, 1], 1)).toBe(true);
+    expect(ColorSpace.isDefaultDecode([1, 0], 1)).toBe(false);
+    expect(ColorSpace.isDefaultDecode([1, 1], 1)).toBe(false);
+    expect(ColorSpace.isDefaultDecode([0, 1, 0, 1, 0, 1, 1, 1], 4)).toBe(false);
```

Comment:
Did you mean for one of the last three cases to be truthy instead, since otherwise I'm not sure why you need so many (almost) identical test-cases?